extensions/security/runtime-spi/src/main/java/io/quarkus/security/spi/runtime/MethodDescription.java

```diff
@@ -0,0 +1,61 @@
+package io.quarkus.security.spi.runtime;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class MethodDescription {
+    private final String className;
+    private final String methodName;
+    private final String[] parameterTypes;
+
+    public MethodDescription(String className, String methodName, String[] parameterTypes) {
+        this.className = className;
+        this.methodName = methodName;
+        this.parameterTypes = parameterTypes;
+    }
+
+    public String getClassName() {
+        return className;
+    }
+
+    public String getMethodName() {
+        return methodName;
+    }
+
+    public String[] getParameterTypes() {
+        return parameterTypes;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+        MethodDescription that = (MethodDescription) o;
+        return className.equals(that.className) &&
+                methodName.equals(that.methodName) &&
+                Arrays.equals(parameterTypes, that.parameterTypes);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = Objects.hash(className, methodName);
+        result = 31 * result + Arrays.hashCode(parameterTypes);
+        return result;
```

Comment:
If I were you, I would cache the hashcode and compute it in the constructor.

Reply:
I just copied what was already there, but yeah, that's a very good idea.

Reply:
Doen